src/Agent.Sdk/Util/ParentProcessUtil/InteropParentProcessFinder.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace Agent.Sdk.Util.ParentProcessUtil
+{
+    internal class Interop
+    {
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct PROCESS_BASIC_INFORMATION
+        {
+            internal IntPtr ExitStatus;
+            internal IntPtr PebBaseAddress;
+            internal IntPtr AffinityMask;
+            internal IntPtr BasePriority;
+            internal IntPtr UniqueProcessId;
+            internal IntPtr InheritedFromUniqueProcessId;
+        }
+
+        [DllImport("ntdll.dll", SetLastError = true)]
```

Comment:
I don't think "SetLastError" is correct; https://learn.microsoft.com/en-us/windows/win32/api/winternl/nf-winternl-ntqueryinformationprocess -- it doesn't say that LastError is set.

SetLastError should only be used if the function uses 'lasterror' via GetLastError to return the error

Reply:
Removing interop code for now, but noted for future if we decide to bring it back